x-pack/legacy/plugins/lens/public/xy_visualization_plugin/xy_expression.tsx

```diff
@@ -127,66 +117,59 @@ export function XYChart({ data, args }: XYChartProps) {
     );
   }
 
-  const { legend, x, y, splitSeriesAccessors, seriesType } = args;
-  // TODO: Stop mapping data once elastic-charts allows axis naming
-  // https://github.com/elastic/elastic-charts/issues/245
-  const seriesProps = {
-    splitSeriesAccessors,
-    stackAccessors: seriesType.includes('stacked') ? [x.accessor] : [],
-    id: getSpecId(y.labels.join(',')),
-    xAccessor: x.accessor,
-    yAccessors: y.labels,
-    data: data.rows.map(row => {
-      const newRow: typeof row = {};
-
-      // Remap data to { 'Count of documents': 5 }
-      Object.keys(row).forEach(key => {
-        const labelIndex = y.accessors.indexOf(key);
-        if (labelIndex > -1) {
-          newRow[y.labels[labelIndex]] = row[key];
-        } else {
-          newRow[key] = row[key];
-        }
-      });
-      return newRow;
-    }),
-  };
```

Comment:
The logic here needs to be copied into each layer

Reply:
Except that we don't currently pass in the operations to the expression, which needs to be done for this to work

Reply:
Fixed